geode-core/src/main/java/org/apache/geode/internal/cache/TxCallbackEventFactoryImpl.java

```diff
@@ -127,13 +126,21 @@ public EntryEventImpl createCallbackEvent(final InternalRegion internalRegion,
           }
         }
       } else if (computeFilterInfo) { // not a bucket
-        if (TxCallbackEventFactoryImpl.logger.isTraceEnabled()) {
-          TxCallbackEventFactoryImpl.logger.trace("createCBEvent computing routing for non-bucket");
-        }
-        FilterProfile fp = internalRegion.getFilterProfile();
-        if (fp != null) {
-          retVal.setLocalFilterInfo(fp.getLocalFilterRouting(retVal));
+        if (!isOriginRemote) {
```

Comment:
Can both condition be replaced with AND condition. And move the comments from bottom to top...